### PR TITLE
Reduced totient (Carmichael's lambda function)

### PR DESCRIPTION
**Pull request by postvakje**
The reduced totient of n is the smallest m > 0 such that for all a coprime to n,
a<sup>m</sup> &equiv; 1 mod n

### CODE REVIEW

**Comment by asmeurer** [comment]
Can you add the definition to the docstring

**Comment by asmeurer** [comment]
And Wikipedia and MathWorld references.

**Comment by asmeurer** [comment]
Other than that this looks good (assuming the algorithm is correct).

**Comment by asmeurer** [comment]
Checked Wikipedia and the algorithm does look correct.

**Comment by postvakje** [comment]
Added descriptions and references to docstrings of both `reduced_totient` and `totient`.

**Comment by postvakje** [comment]
please review

**Comment by asmeurer** [comment]
There are merge conflicts.

File: sympy/ntheory/factor_.py [hunk 1]
```diff
@@ -1529,6 +1529,18 @@ class totient(Function):
     """
     Calculate the Euler totient function phi(n)
 
+    ``totient(n)`` or `\phi(n)` is the number of positive integers `\leq` n
```
Comment (asmeurer) [inline sympy/ntheory/factor_.py]: Should be one back quote for latex math.

**Comment by postvakje** [comment]
Merged with latest master and fixed merge conflicts.  I think all latex math have single backquotes.

**Comment by postvakje** [comment]
Please review.

**Pull request merged**